### ISSUE TITLE
fuser tcp, udp tests fail inconsistently, on Linux

### ISSUE THREAD

**Issue by jgarzik** [rustcoreutils/posixutils-rs#240]
fuser tests do not succeed 100% of the time, on linux.  Failure rates are higher on Linux, but still present also on macos.

On linux, the `test_fuser_udp` and `test_fuser_tcp` tests frequently fail.

cc @Wandalen

**Comment by jgarzik** [comment]
Example Linux failure:
```

     Running tests/process-tests.rs (target/debug/deps/process_tests-aacbf1bf73362fd7)

running 14 tests
test fuser::test_fuser_with_many_files ... ignored
test xargs::xargs_trace ... ok
test xargs::xargs_basic ... ok
test xargs::xargs_with_maxnum ... ok
test xargs::xargs_with_eofstr ... ok
test xargs::xargs_with_null_delimiter ... ok
test xargs::xargs_with_maxsize ... ok
test xargs::xargs_with_null_delimiter_trailing_non_null ... ok
test fuser::test_fuser_basic ... ok
test fuser::test_fuser_unixsocket ... ok
test fuser::test_fuser_udp ... ok
test fuser::test_fuser_with_user ... ok
test fuser::test_fuser_tcp ... FAILED
test fuser::test_fuser_large_directory ... ok

failures:

---- fuser::test_fuser_tcp stdout ----
thread 'fuser::test_fuser_tcp' panicked at process/tests/fuser/mod.rs:205:9:
assertion `left == right` failed
  left: Some(1)
 right: Some(0)
stack backtrace:
   0: rust_begin_unwind
             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/std/src/panicking.rs:665:5
   1: core::panicking::panic_fmt
             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/core/src/panicking.rs:74:14
   2: core::panicking::assert_failed_inner
             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/core/src/panicking.rs:410:17
   3: core::panicking::assert_failed
             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/core/src/panicking.rs:365:5
   4: process_tests::fuser::test_fuser_tcp::{{closure}}::{{closure}}
             at ./tests/fuser/mod.rs:205:9
   5: plib::testing::run_test_with_checker
             at /home/jgarzik/repo/posixutils-rs/plib/src/testing.rs:111:5
   6: process_tests::fuser::fuser_test
             at ./tests/fuser/mod.rs:22:5
   7: process_tests::fuser::test_fuser_tcp::{{closure}}
             at ./tests/fuser/mod.rs:203:5
   8: <core::pin::Pin<P> as core::future::future::Future>::poll
             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/core/src/future/future.rs:123:9
   9: <core::pin::Pin<P> as core::future::future::Future>::poll
             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/core/src/future/future.rs:123:9
  10: tokio::runtime::scheduler::current_thread::CoreGuard::block_on::{{closure}}::{{closure}}::{{closure}}
             at /home/jgarzik/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.40.0/src/runtime/scheduler/current_thread/mod.rs:696:57
  11: tokio::runtime::coop::with_budget
             at /home/jgarzik/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.40.0/src/runtime/coop.rs:107:5
  12: tokio::runtime::coop::budget
             at /home/jgarzik/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.40.0/src/runtime/coop.rs:73:5
  13: tokio::runtime::scheduler::current_thread::CoreGuard::block_on::{{closure}}::{{closure}}
             at /home/jgarzik/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.40.0/src/runtime/scheduler/current_thread/mod.rs:696:25
  14: tokio::runtime::scheduler::current_thread::Context::enter
             at /home/jgarzik/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.40.0/src/runtime/scheduler/current_thread/mod.rs:423:19
  15: tokio::runtime::scheduler::current_thread::CoreGuard::block_on::{{closure}}
             at /home/jgarzik/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.40.0/src/runtime/scheduler/current_thread/mod.rs:695:36
  16: tokio::runtime::scheduler::current_thread::CoreGuard::enter::{{closure}}
             at /home/jgarzik/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.40.0/src/runtime/scheduler/current_thread/mod.rs:774:68
  17: tokio::runtime::context::scoped::Scoped<T>::set
             at /home/jgarzik/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.40.0/src/runtime/context/scoped.rs:40:9
  18: tokio::runtime::context::set_scheduler::{{closure}}
             at /home/jgarzik/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.40.0/src/runtime/context.rs:180:26
  19: std::thread::local::LocalKey<T>::try_with
             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/std/src/thread/local.rs:283:12
  20: std::thread::local::LocalKey<T>::with
             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/std/src/thread/local.rs:260:9
  21: tokio::runtime::context::set_scheduler
             at /home/jgarzik/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.40.0/src/runtime/context.rs:180:9
  22: tokio::runtime::scheduler::current_thread::CoreGuard::enter
             at /home/jgarzik/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.40.0/src/runtime/scheduler/current_thread/mod.rs:774:27
  23: tokio::runtime::scheduler::current_thread::CoreGuard::block_on
             at /home/jgarzik/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.40.0/src/runtime/scheduler/current_thread/mod.rs:683:19
  24: tokio::runtime::scheduler::current_thread::CurrentThread::block_on::{{closure}}
             at /home/jgarzik/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.40.0/src/runtime/scheduler/current_thread/mod.rs:191:28
  25: tokio::runtime::context::runtime::enter_runtime
             at /home/jgarzik/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.40.0/src/runtime/context/runtime.rs:65:16
  26: tokio::runtime::scheduler::current_thread::CurrentThread::block_on
             at /home/jgarzik/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.40.0/src/runtime/scheduler/current_thread/mod.rs:179:9
  27: tokio::runtime::runtime::Runtime::block_on_inner
             at /home/jgarzik/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.40.0/src/runtime/runtime.rs:361:47
  28: tokio::runtime::runtime::Runtime::block_on
             at /home/jgarzik/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.40.0/src/runtime/runtime.rs:335:13
  29: process_tests::fuser::test_fuser_tcp
             at ./tests/fuser/mod.rs:203:5
  30: process_tests::fuser::test_fuser_tcp::{{closure}}
             at ./tests/fuser/mod.rs:201:26
  31: core::ops::function::FnOnce::call_once
             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/core/src/ops/function.rs:250:5
  32: core::ops::function::FnOnce::call_once
             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/core/src/ops/function.rs:250:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.


failures:
    fuser::test_fuser_tcp

test result: FAILED. 12 passed; 1 failed; 1 ignored; 0 measured; 0 filtered out; finished in 0.91s
```

**Comment by jgarzik** [comment]
See #246 for macos' fuser test failure.

**Comment by jgarzik** [comment]
appears to be fixed after #261 (and #264)
